extensions/big-data-cluster/src/extension.ts

```diff
@@ -59,6 +65,52 @@ function registerCommands(context: vscode.ExtensionContext, treeDataProvider: Co
 		const dashboard: BdcDashboard = new BdcDashboard(title, new BdcDashboardModel(node.url, node.auth, node.username, node.password));
 		dashboard.showDashboard();
 	});
+
+	vscode.commands.registerCommand(MountHdfsCommand, e => mountHdfs(e).catch(error => {
+		vscode.window.showErrorMessage(error instanceof Error ? error.message : error);
+	}));
+}
+
+async function mountHdfs(explorerContext?: azdata.ObjectExplorerContext): Promise<void> {
+	let endpoint = await lookupController(explorerContext);
+	if (!endpoint) {
+		vscode.window.showErrorMessage(endpointNotFoundError);
+		return;
+	}
+	let profile = explorerContext.connectionProfile;
+	let mountProps: MountHdfsProperties = {
+		url: endpoint,
+		auth: profile.authenticationType === 'SqlLogin' ? 'basic' : 'integrated',
+		username: profile.userName,
+		password: profile.password,
+		hdfsPath: getHdsfPath(explorerContext.nodeInfo.nodePath)
+	};
+	let dialog = new MountHdfsDialog(new MountHdfsDialogModel(mountProps));
+	dialog.showDialog();
+}
+
+function getHdsfPath(nodePath: string): string {
+	const hdfsNodeLabel = '/HDFS';
```

Comment:
>HDFS [](start = 25, length = 4)

This shouldn't be necessary if you get the actual HdfsFileSourceNode (that has a hdfsPath property on it). See how hdfsCommands are doing it

Reply:
Ah shoot didn't notice that this was in the bdc extension. Hmmmm, there's gotta be a better way. The nodeInfo doesn't have anything useful on it?

---
In reply to: [332774950](https://github.com/microsoft/azuredatastudio/pull/7580#discussion_r332774950) [](ancestors = 332774950)

Reply:
No, it unfortunately doesn't have anything better (that I'm aware of, anyhow). So... unless I expose a requirement on the mssql extension which also isn't great, I think this is the "best" alternative. We know HDFS is a non-localized string so we can look for it and split, it works "ok"

Reply:
One thing we should strongly consider doing is exposing functionality like this through extension APIs. We could definitely add an export to the mssql extension that gets a node path if that's something that's needed. (Although it would probably be better in this case to modify the azdata api with that functionality)

Reply:
I don't want a hard binding here, but would like built-in functionality to provide this in some common field?